File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.kt
class Solution {
    fun solution(k: Int, tangerine: IntArray): Int {
        // 1️⃣ 크기별 개수 세기
        val countMap = mutableMapOf<Int, Int>()
        for (t in tangerine) {
            countMap[t] = countMap.getOrDefault(t, 0) + 1
        }

        // 2️⃣ 개수만 추출해서 내림차순 정렬
        val sortedCounts = countMap.values.sortedDescending()

        // 3️⃣ 큰 개수부터 더하면서 k개 채우기
        var total = 0
        var answer = 0
        for (c in sortedCounts) {
            total += c
            answer++
            if (total >= k) break
        }

        return answer
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    counts = Counter(tangerine)          # 귤 크기별 개수 세기
    sorted_counts = sorted(counts.values(), reverse=True)  # 개수 내림차순 정렬
    
    answer = 0
    total = 0
    
    for count in sorted_counts:
        total += count      # 현재 크기 귤 더하기
        answer += 1         # 사용한 크기 종류 +1
        if total >= k:      # k개 이상 채우면 종료
            break
            
    return answer
